packages/atomic/src/components/common/item-list/item-list-common.spec.ts

```diff
@@ -0,0 +1,301 @@
+import {CommerceStore} from '@/src/components';
+import {getFirstFocusableDescendant} from '@/src/utils/accessibility-utils';
+import {updateBreakpoints} from '@/src/utils/replace-breakpoint';
+import {beforeEach, describe, expect, it, vi} from 'vitest';
+import {ItemListCommonProps} from './item-list-common';
+import {ItemListCommon} from './item-list-common';
+
+vi.mock('@/src/utils/accessibility-utils', {spy: true});
+vi.mock('@/src/utils/replace-breakpoint', {spy: true});
+
+describe('ItemListCommon', () => {
+  describe('#constructor', () => {
+    it('should create an instance', () => {
+      const itemListCommon = itemListCommonFixture();
+
+      expect(itemListCommon).toBeInstanceOf(ItemListCommon);
+    });
+
+    it('should call #props.store.setLoadingFlag with #props.loadingFlag', () => {
+      const setLoadingFlag = vi.fn();
+
+      itemListCommonFixture({
+        loadingFlag: 'test-loading-flag',
+        store: fakeStore({setLoadingFlag}),
+      });
+
+      expect(setLoadingFlag).toHaveBeenCalledOnce();
+      expect(setLoadingFlag).toHaveBeenCalledWith('test-loading-flag');
+    });
+
+    it('should set #props.store.state.resultList to the instance', () => {
+      const store = fakeStore();
+
+      const itemListCommon = itemListCommonFixture({store});
+
+      expect(store.state.resultList).toBe(itemListCommon);
+    });
+  });
+
+  describe('#updateBreakpoints', () => {
+    it('should call the #updateBreakpoints util function with #props.host only once', () => {
+      const mockedUpdateBreakpoints = vi.mocked(updateBreakpoints);
+      const host = document.createElement('span');
+
+      const itemListCommon = itemListCommonFixture({host});
+
+      expect(mockedUpdateBreakpoints).not.toHaveBeenCalled();
+
+      itemListCommon.updateBreakpoints();
+      itemListCommon.updateBreakpoints();
+      itemListCommon.updateBreakpoints();
```

Comment:
I understand we "mash" three, but unless there is specific logic related to 3 calls, I think 2 is enough and reflects the expected behavior better.